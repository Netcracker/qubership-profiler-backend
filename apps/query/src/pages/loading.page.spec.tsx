import LoadingPage from '@app/pages/loading.page';
import { render, screen } from '@testing-library/react';

test('should display text when it passed', () => {
    render(<LoadingPage text="text of loading" />);

    expect(screen.getByText('text of loading')).toBeInTheDocument();
});

test('should change text on rerender', () => {
    render(<LoadingPage text="text of loading" />);

    expect(screen.getByText('text of loading')).toBeInTheDocument();

    render(<LoadingPage text="changed text" />);

    expect(screen.getByText('changed text')).toBeInTheDocument();
});

test('should match snapshot', () => {
    const { container } = render(<LoadingPage text="text of loading" />);

    expect(container).toMatchSnapshot();
});
