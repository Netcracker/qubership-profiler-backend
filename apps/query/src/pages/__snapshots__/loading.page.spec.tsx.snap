// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-spinning ant-spin-show-text loader-module_loader__fc32e5 taLoader taActive"
    >
      <svg
        aria-label="loading icon"
        class="loader-module_circleLoader__fc32e5 ant-spin-dot"
        focusable="false"
        role="img"
        style="font-size: 48px;"
        viewBox="0 0 50 50"
      >
        <circle
          cx="25"
          cy="25"
          r="20"
        />
      </svg>
      <div
        class="ant-spin-text"
      >
        text of loading
      </div>
    </div>
  </div>
</div>
`;
